src/dfx/src/commands/info/candid_ui_url.rs

```diff
@@ -0,0 +1,38 @@
+use crate::commands::deploy::MAINNET_CANDID_INTERFACE_PRINCIPAL;
+use crate::lib::{error::DfxResult, named_canister::UI_CANISTER};
+use crate::Environment;
+use anyhow::{bail, Context};
+use url::{Host::Domain, Url};
+
+pub(crate) fn get_candid_ui_url(env: &dyn Environment) -> DfxResult<String> {
```

Comment:
Could we DRY this with construct_ui_canister_url [in deploy.rs](https://github.com/dfinity/sdk/blob/master/src/dfx/src/commands/deploy.rs#L320)?